src/components/side_panel/data_validation/dv_criterion_form/dv_date_criterion/dv_date_criterion.ts

```diff
@@ -0,0 +1,49 @@
+import { useState } from "@odoo/owl";
+import { _t } from "../../../../../translation";
+import { DataValidationDateCriterion, DateCriterionValue } from "../../../../../types";
+import { DataValidationCriterionForm } from "../dv_criterion_form";
+import { DataValidationInput } from "../dv_input/dv_input";
+
+export const DATES_VALUES: Record<DateCriterionValue, string> = {
+  today: _t("today"),
+  yesterday: _t("yesterday"),
+  tomorrow: _t("tomorrow"),
+  lastWeek: _t("in the past week"),
+  lastMonth: _t("in the past month"),
+  lastYear: _t("in the past year"),
+  exactDate: _t("exact date"),
+};
+
+export class DataValidationDateCriterionForm extends DataValidationCriterionForm<DataValidationDateCriterion> {
+  static template = "o-spreadsheet-DataValidationDateCriterion";
+  static components = { DataValidationInput };
+
+  state = useState({ dateValue: this.props.initialCriterion.dateValue || "exactDate" });
+
+  setup() {
+    super.setup();
+    this.criterion = {
+      ...this.props.initialCriterion,
+      dateValue: this.props.initialCriterion.dateValue || "exactDate",
+    };
+  }
+
+  onValueChanged(ev: Event) {
+    this.criterion.values[0] = (ev.target as HTMLInputElement).value;
+    this.updateCriterion();
+  }
+
+  onDateValueChanged(ev: Event) {
+    const dateValue = (ev.target as HTMLInputElement).value as DateCriterionValue;
+    this.state.dateValue = dateValue;
+    this.criterion.dateValue = dateValue;
+    this.updateCriterion();
+  }
+
+  get dateValues() {
+    return Object.keys(DATES_VALUES).map((key) => ({
```

Comment:
I remember why I did it that way, it is so that typescript will yell at us if we don't add a key in DATES_VALUES when adding something in DateCriterionValue. Maybe we don't care ?